src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java

```diff
@@ -2,51 +2,45 @@
 
 import java.io.IOException;
 import java.net.ServerSocket;
-import java.net.Socket;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.logging.Level;
 
-import static com.spanish_inquisition.battleship.common.AppLogger.DEFAULT_LEVEL;
 import static com.spanish_inquisition.battleship.common.AppLogger.initializeLogger;
 import static com.spanish_inquisition.battleship.common.AppLogger.logger;
 
 public class BattleshipServer {
     private static final int NUMBER_OF_PLAYERS = 2;
     private static final Integer PORT_NUMBER = 6666;
-    private static List<Socket> clients;
+    static List<ClientConnectionHandler> clients;
 
     public static void main(String[] args) {
         initializeLogger();
         ServerSocket serverSocket = createServerSocket();
-        clients = connectWithPlayers(serverSocket);
         connectWithPlayers(serverSocket);
     }
 
-    static List<Socket> connectWithPlayers(ServerSocket serverSocket) {
-        List<Socket> clientSockets = new ArrayList<>();
+    static void connectWithPlayers(ServerSocket serverSocket) {
+        List<ClientConnectionHandler> clientsHandlers = new ArrayList<>();
         for (int i = 0; i < NUMBER_OF_PLAYERS; i++) {
-            new Thread(()->{
-                logger.log(DEFAULT_LEVEL, "Hello from thread " + Thread.currentThread().getName());
-                Socket clientSocket = null;
-                try {
-                    clientSocket = serverSocket.accept();
-                } catch (IOException e) {
-                    logger.log(DEFAULT_LEVEL, "There was an error while waiting for connection");
-                }
-                clientSockets.add(clientSocket);
-            }).run();
+            ClientConnectionHandler clientConnectionHandler = null;
+            try {
+                clientConnectionHandler = new ClientConnectionHandler(serverSocket);
+                clientConnectionHandler.start();
+            } catch (IOException e) {
+                logger.log(Level.WARNING, "couldn't accept connection from client");
```

Comment:
Add an Exception object as a third parameter  to the logging method, it will print out the stack traces in log file